app/src/main/java/com/droidhats/campuscompass/models/Location.kt

```diff
@@ -0,0 +1,109 @@
+package com.droidhats.campuscompass.models
+
+import android.util.Log
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Polygon
+import com.google.android.gms.maps.model.PolygonOptions
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/*
+* Model for location classes
+* Data relating to locations should be stored in this file
+* */
+
+abstract class Location(coordinate: LatLng) {
+    private var coordinate: LatLng = coordinate
+}
+
+// Model for Campus class
+class Campus(coordinate: LatLng, name: String, jsonObject: JSONObject) : Location(coordinate) {
+    private var buildingsList: MutableList<Building> = mutableListOf()
+    private val name: String = name
+    private val jsonObject: JSONObject = jsonObject
+    fun getName(): String = name
+    fun getBuildings(): MutableList<Building> = buildingsList
+
+    init{
+        createBuildings()
+    }
+
+    private fun createBuildings() {
+        try{
+            var buildingsArray : JSONArray = when (name) {
+                "SGW" -> { //Important that at the creation of the campus object, its name is either SGW or Loyola; otherwise the parsing fails
+                    jsonObject.getJSONArray("SGW_buildings")
+                }
+                "Loyola" -> {
+                    jsonObject.getJSONArray("LOY_buildings")
+                }
+                else -> {
+                    Log.v("Parsing error", "Unable to parse buildings from JSON\nMake sure that at the creation of the campus object the name parameter is either SGW or Loyola)" +
+                            "\nMake sure that the values of the string resources SGW_Campus_Name and Loyola_Campus_Name are 'SGW' and 'Loyola'")
+                    return
+                }
+            }
+
+            var coordinatesArray : JSONArray
+
+            //Traverse each building in the array
+            for(i in 0 until buildingsArray.length()){
+                val buildingName : String = buildingsArray.getJSONObject(i).get("name").toString()
+                val buildingLocationArray: JSONArray = buildingsArray.getJSONObject(i).getJSONArray("location")
+                val buildingLocation = LatLng(buildingLocationArray[0].toString().toDouble(), buildingLocationArray[1].toString().toDouble())
+
+                //Create the building object
+                var building = Building(buildingLocation, buildingName)
```

Comment:
I feel like we should be passing to the construct the list of coordinates instead of necessitating a mutable list in the building class to append coordinates to later, since we will never really change the values of the coordinates once it's instantiated.

Reply:
fixed. will commit changes now